src/Microsoft.IdentityModel.Logging/LogHelper.cs

```diff
@@ -36,20 +37,109 @@ namespace Microsoft.IdentityModel.Logging
     public class LogHelper
     {
         /// <summary>
-        /// Logs an event using the event source logger and throws the exception.
+        /// Logs an event using the event source logger and returns new <see cref="ArgumentNullException"/> exception.
+        /// </summary>
+        /// <param name="argument">argument that is null or empty.</param>
+        public static ArgumentNullException LogArgumentNullException(string argument)
+        {
+            return LogException<ArgumentNullException>(EventLevel.Error, null, "IDX10000: The parameter '{0}' cannot be a 'null' or an empty object.", argument);
```

Comment:
Yeah right. There is one already in the LogMessages. It uses the prefix MIML...but thats ok.